### ISSUE TITLE
Inconsistent help about subscription option

### ISSUE THREAD

**Issue by Alessar** [Azure/azure-xplat-cli#3417]
CLI Version: **0.10.6**
OS Type: **Win**
Mode: **ARM**

Help about any command contains required option `--subscription <subscription>`
But commands without that option work normally as well.
So, maybe option **subscription** should be marked as optional in help like `--subscription [subscription]` ?

**Comment by amarzavery** [comment]
I agree with you on this. The CLI takes the default subscription from the profile if it is not provided. It is a low hanging fruit for us right now. We are willing to accept PRs.
